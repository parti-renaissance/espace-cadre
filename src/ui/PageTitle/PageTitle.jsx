import { Grid, Typography } from '@mui/material'
import { Link } from 'react-router-dom'
import PropTypes from 'prop-types'

const PageTitle = ({
  title,
  titleLink,
  titleSuffix,
  breakpoints = { xs: null, sm: null, md: null, lg: null, xl: null },
}) => (
  <Grid
    item
    sx={{ fontSize: 24 }}
    xs={breakpoints.xs}
    sm={breakpoints.sm}
    md={breakpoints.md}
    lg={breakpoints.lg}
    xl={breakpoints.xl}
  >
    {titleLink ? (
      <Typography variant="h4">
        <Link to={titleLink}>{title}</Link>
      </Typography>
    ) : (
      <Typography variant="h4">{title}</Typography>
    )}
    {titleSuffix && (
      <>
        <Typography variant="pageTitle" sx={{ color: 'gray400' }}>
          &nbsp;{'>'}&nbsp;
        </Typography>
        <Typography variant="pageTitle" sx={{ color: 'gray800' }}>
          {titleSuffix}
        </Typography>
      </>
    )}
  </Grid>
)

export default PageTitle

PageTitle.propTypes = {
  title: PropTypes.string.isRequired,
  titleLink: PropTypes.string,
  titleSuffix: PropTypes.string,
  breakpoints: PropTypes.shape({
    xs: PropTypes.number,
    sm: PropTypes.number,
    md: PropTypes.number,
    lg: PropTypes.number,
    xl: PropTypes.number,
  }),
}
