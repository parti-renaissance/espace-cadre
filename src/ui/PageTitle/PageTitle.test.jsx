import { render } from '@testing-library/react'
import PageTitle from './PageTitle'

vi.mock('@mui/material', () => ({
  Grid: ({ children }) => <div>{children}</div>,
  Typography: ({ children }) => <div className="typography">{children}</div>,
}))
vi.mock('react-router-dom', () => ({
  Link: ({ to, children }) => (
    <div className="Link" data-to={to}>
      {children}
    </div>
  ),
}))

describe('PageTitle', () => {
  it('displays only a title', () => {
    const { container } = render(<PageTitle title="Title for test" breakpoints={{ xs: 0 }} />)

    expect(container).toMatchSnapshot()
  })

  it('displays a breadcrumbs', () => {
    const { container } = render(
      <PageTitle title="Title for test" titleLink="titleLink" titleSuffix="titleSuffix" breakpoints={{ xs: 0 }} />
    )

    expect(container).toMatchSnapshot()
  })
})
