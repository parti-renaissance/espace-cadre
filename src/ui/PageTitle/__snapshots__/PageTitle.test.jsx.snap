// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`PageTitle > displays a breadcrumbs 1`] = `
<div>
  <div>
    <div
      class="typography"
    >
      <div
        class="Link"
        data-to="titleLink"
      >
        Title for test
      </div>
    </div>
    <div
      class="typography"
    >
       
      &gt;
       
    </div>
    <div
      class="typography"
    >
      titleSuffix
    </div>
  </div>
</div>
`;

exports[`PageTitle > displays only a title 1`] = `
<div>
  <div>
    <div
      class="typography"
    >
      Title for test
    </div>
  </div>
</div>
`;
